### PR TITLE
Improve Dockerfiles for Linux

### PR DESCRIPTION
**Pull request by izzyleung**
Overall improvements to `Dockerfile`s for both agent and server.
Regenerate documentation.
Reduced overall image sizes.

### CODE REVIEW

File: configs/linux/Agent/Ubuntu/Ubuntu.Dockerfile [hunk 1]
```diff
@@ -43,7 +43,7 @@ ENV DOTNET_CLI_TELEMETRY_OPTOUT=true \
 ARG dotnetCoreLinuxComponent
 
 RUN apt-get update && \
-    apt-get install -y git mercurial apt-transport-https ca-certificates software-properties-common && \
```
Comment (izzyleung) [inline configs/linux/Agent/Ubuntu/Ubuntu.Dockerfile:46]: `ca-certificates` was installed in `configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile`, L#22.

File: configs/linux/Agent/Ubuntu/Ubuntu.Dockerfile [hunk 2]
```diff
@@ -58,7 +58,6 @@ RUN apt-get update && \
     curl -SL "https://github.com/docker/compose/releases/download/1.24.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose && chmod +x /usr/local/bin/docker-compose && \
     \
     apt-get install -y --no-install-recommends \
-            fontconfig \
```
Comment (izzyleung) [inline configs/linux/Agent/Ubuntu/Ubuntu.Dockerfile:61]: `fontconfig` was installed in `configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile`, L#22.

File: configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile [hunk 3]
```diff
@@ -18,11 +18,12 @@ FROM ${ubuntuImage}
 
 ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
 
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends curl ca-certificates fontconfig locales unzip \
-    && echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen \
-    && locale-gen en_US.UTF-8 \
-    && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && \
```
Comment (izzyleung) [inline configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile:21]: Unify code style.

File: configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile [hunk 4]
```diff
@@ -52,30 +53,23 @@ ENV CONFIG_FILE=/data/teamcity_agent/conf/buildAgent.properties \
...
-COPY run-agent-services.sh /run-services.sh
-COPY TeamCity/buildAgent /opt/buildagent
+COPY --chown=buildagent:buildagent run-agent.sh /run-agent.sh
+COPY --chown=buildagent:buildagent run-agent-services.sh /run-services.sh
+COPY --chown=buildagent:buildagent TeamCity/buildAgent /opt/buildagent
 
-RUN apt-get update && \
-    useradd -m buildagent && \
-    chmod +x /opt/buildagent/bin/*.sh && \
+RUN chmod +x /opt/buildagent/bin/*.sh && \
```
Comment (izzyleung) [inline configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile:60]: By combining those two `RUN` commands, the number of layers in the final image will reduce by one.

File: configs/linux/Server/Ubuntu/Ubuntu.Dockerfile [hunk 5]
```diff
@@ -58,7 +58,7 @@ LABEL dockerImage.teamcity.version="latest" \
       dockerImage.teamcity.buildNumber="latest"
 
 RUN apt-get update && \
-    apt-get install -y git mercurial ca-certificates && \
```
Comment (izzyleung) [inline configs/linux/Server/Ubuntu/Ubuntu.Dockerfile:61]: `ca-certificates` was installed in this very file at L#22.

File: configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile [hunk 6]
```diff
@@ -52,30 +53,23 @@ ENV CONFIG_FILE=/data/teamcity_agent/conf/buildAgent.properties \
...
       dockerImage.teamcity.buildNumber="latest"
 
-COPY run-agent.sh /run-agent.sh
-COPY run-agent-services.sh /run-services.sh
-COPY TeamCity/buildAgent /opt/buildagent
+COPY --chown=buildagent:buildagent run-agent.sh /run-agent.sh
+COPY --chown=buildagent:buildagent run-agent-services.sh /run-services.sh
+COPY --chown=buildagent:buildagent TeamCity/buildAgent /opt/buildagent
 
-RUN apt-get update && \
```
Comment (izzyleung) [inline configs/linux/MinimalAgent/Ubuntu/Ubuntu.Dockerfile:59]: Don't think the `apt-get update` here is necessary, removing it.

**Comment by NikolayPianikov** [comment]
@izzyleung thank you

**Pull request merged**